src/app/hearing-part/components/listing-create/listing-create.component.html

```diff
@@ -36,10 +36,39 @@
                     </mat-select>
                 </mat-form-field>
 
-                <mat-form-field>
-                    <input id="duration" formControlName="duration" [ngModel]="listing.hearing.duration | appDurationAsMinutes" (ngModelChange)="updateDuration($event)" type="number" matInput placeholder="Estimated duration *" min="1" >
-                    <span matSuffix>Minutes</span>
-                </mat-form-field>
+                <div>
+                <div for="session-mode-rg" class="mat-form-field-wrapper">Choose session mode *</div>
+                <div class="mat-form-field-wrapper">
+                    <mat-radio-group id="session-mode-rg"
+                                     formControlName="listingTypeRadio"
+                                     (change)="onListingTypeChange($event)"
+                                     required>
+                        <mat-radio-button id="single-session-radio"  value="0" [disabled]="this.editMode && this.isMultiSession" >Single session</mat-radio-button>
+                        <mat-radio-button id="multi-session-radio"  value="1" [disabled]="this.editMode && this.isMultiSession" >Multi session</mat-radio-button>
+                    </mat-radio-group>
+                </div>
+
+                <div formGroupName="listingType">
+                    <div *ngIf="chosenListingType === 0">
+                        <mat-form-field>
+                            <input id="duration-minutes" formControlName="duration" [value]="listing.hearing.duration | appDurationAsMinutes" (ngModelChange)="updateDuration($event, 'minute')" type="number" matInput placeholder="Estimated duration *" min="1" max="1439" step="1" >
+                            <span matSuffix>Minutes</span>
+                        </mat-form-field>
+                    </div>
+
+                    <div *ngIf="chosenListingType === 1">
```

Comment:
there's enum defined in component, but it's not used here, use enum

Reply:
that was original thinking, but to make it work I would have to use this approach https://stackoverflow.com/a/44045373 or stay with what is here

Reply:
when I am reading 'chosenListingType === 1' I have no idea what that means. Maybe we can create two functions in component that can checks? iSingleSessionSelected, isMultiSessionSelected?

Reply:
ok